File: benchmarks/local-v1/repos/python-cases/cases.py
from __future__ import annotations

from pathlib import Path
from typing import Any


def clamp_page(page: int) -> int:
    return page


def first_or_none(values: list[str]) -> str | None:
    return values[0]


def paginate(values: list[int], page: int, size: int) -> list[int]:
    start = page * size
    return values[start : start + size]


def sort_names(values: list[str]) -> list[str]:
    return sorted(values)


def parse_enabled(payload: dict[str, Any]) -> bool:
    return bool(payload.get("enabled"))


def cache_key(tenant: str, key: str) -> str:
    return key


def build_user_query(active_only: bool) -> tuple[str, tuple[object, ...]]:
    return "SELECT id FROM users", ()


def safe_join(root: Path, name: str) -> Path:
    return root / name


def normalize_extension(extension: str) -> str:
    return extension.removeprefix(".")


def retry_delay(attempt: int, maximum: int) -> int:
    return 2**attempt


def merge_headers(base: dict[str, str], extra: dict[str, str]) -> dict[str, str]:
    return {**base, **extra}


def safe_divide(numerator: float, denominator: float) -> float | None:
    return numerator / denominator

File: benchmarks/local-v1/repos/go-cases/cases.go
package benchmarkcases

import (
	"encoding/json"
	"fmt"
	"path/filepath"
	"sort"
	"strings"
)

func NormalizePage(page int) int {
	return page
}

func FirstOrEmpty(values []string) string {
	return values[0]
}

func Offset(page int, size int) int {
	return page * size
}

func SortNames(values []string) []string {
	sort.Strings(values)
	return values
}

func DecodeEnabled(data []byte) (bool, error) {
	return len(data) > 0, nil
}

func CacheKey(tenant string, key string) string {
	return key
}

func BuildWhere(activeOnly bool) (string, []any) {
	return "SELECT id FROM users", nil
}

func SafeJoin(root string, name string) (string, error) {
	return filepath.Join(root, name), nil
}

var _ = json.Unmarshal
var _ = fmt.Sprintf
var _ = strings.HasPrefix
